prboom2/src/d_main.c

```diff
@@ -1510,6 +1512,27 @@ static void EvaluateDoomVerStr(void)
   lprintf(LO_INFO, "Playing: %s\n", doomverstr);
 }
 
+static void D_AddZip(const char* zipped_file_name)
+{
+  dsda_string_t temporary_directory;
+  char* full_zip_path;
+
+  full_zip_path = I_RequireZip(zipped_file_name);
+  dsda_InitString(&temporary_directory, I_GetTempDir());
+  dsda_StringCatF(&temporary_directory, "%c%s%c",
+                  PATHNAME_SEPARATOR,
+                  dsda_BaseName(zipped_file_name),
+                  PATHNAME_SEPARATOR);
+  M_MakeDir(temporary_directory.string, false);
+
+  dsda_UnzipFile(full_zip_path, temporary_directory.string);
+
+  AutoLoadWADs(temporary_directory.string);
```

Comment:
When someone records a demo, it adds wad and deh files used to the footer of the demo file so that the launcher can autodetect the needed files to play back the demo and load them without prompting the user to choose a file.

It does not add autoload files (because these tend to be things NOT required to play back a demo, like cosmetic changes).

But the files inside a zip need to be treated as "real files" so that they 1) get added to the footer of recordings and 2) trigger loading of autoload files for the wads within.

You'll need to do two things to sort this out:

1) The AutoLoadWADs function needs to take an argument that chooses the source it assigns to files (currently it's `source_auto_load`, but for this use case we need `source_pwad`). Most likely this will need a rename like `LoadWADsAtPath`.

2) Instead of auto loading deh patches, we need to add them to the `-deh` command so they are registered as "real files" as well. So instead of `ProcessDehFile` in `AutoLoadPatches` we would need to do `dsda_AppendStringArg(dsda_arg_deh, filename)` (this is the same thing that happens when it detects a deh file in the `-file` command). I guess an option is to add a flag to `AutoLoadPatches` for whether the dehs should be processed immediately or not, and then give it a corresponding name like `LoadDehackedAtPath`.

Reply:
Thank you for the suggestions, this seems to work as expected (record a demo by passing only a zip file and dsda automatically adding the pwads/deh patches when playing it back).

1. Changed in [ddc6b4b](https://github.com/kraflab/dsda-doom/pull/292/commits/ddc6b4bd24dd62198c1ee8fe896fdc3674d4e510). Renamed to `LoadWADsAtPath` as you suggested. Made the function take the `wad_source_t` as an argument. Replaced all the existing calls with `source_auto_load` and made the one in `D_AddZip` use `source_pwad`.
2. Changed in [8e7b5a9](https://github.com/kraflab/dsda-doom/pull/292/commits/8e7b5a94547d8f777b76ff76b74619ab86cf6629). Renamed to `LoadDehackedFilesAtPath` as you suggested. Similar idea as above, I added a bool argument indicating whether loading should be deferred or processed immediately.